micro-rdk-installer/src/main.rs

```diff
@@ -331,3 +367,105 @@ fn main() -> Result<(), Error> {
     };
     Ok(())
 }
+
+fn update_app_image(args: &AppImageArgs) -> Result<(), Error> {
+    let config = Config::load().map_err(|err| Error::SerialConfigError(err.to_string()))?;
+
+    let dir = tempfile::tempdir().map_err(Error::FileError)?;
+    let tmp_old = dir.path().join("running-ptable.img");
```

Comment:
Maybe better to not name the file and just use https://docs.rs/tempfile/latest/tempfile/fn.tempfile.html instead.

Reply:
ok, so it seems `read_flash` needs a `Path` as input and the `File` created from tempfile doesn't have a path. TIL not all `File`s have paths, making them more akin to file references or descriptors.